File: internal/types/enterprise_administration.go
package types

const EnterpriseAdministrationQueueV1 = "EnterpriseAdministrationQueueV1"

type EnterpriseAdministrationFacts struct {
	Role                                string `json:"role"`
	ActiveMemberCount                   int    `json:"active_member_count"`
	OperatingAnalysisMissingAccessCount int    `json:"operating_analysis_missing_access_count"`
}

type EnterpriseAdministrationItem struct {
	Code     string `json:"code"`
	Priority string `json:"priority"`
	Count    int    `json:"count"`
	Target   string `json:"target"`
}

type EnterpriseAdministrationPlatformSummary struct {
	ServiceLevel string `json:"service_level"`
	Status       string `json:"status"`
	MemberQuota  *int   `json:"member_quota"`
	Health       string `json:"health"`
}

type EnterpriseAdministrationPlatformProjection struct {
	ContractVersion string                                  `json:"contract_version"`
	Scope           PlatformSettingsScope                   `json:"scope"`
	AsOf            string                                  `json:"as_of"`
	Summary         EnterpriseAdministrationPlatformSummary `json:"summary"`
	Items           []EnterpriseAdministrationItem          `json:"items"`
}

type EnterpriseAdministrationSummary struct {
	EnterpriseAdministrationPlatformSummary
	MemberUsage      int   `json:"member_usage"`
	StorageUsageByte int64 `json:"storage_usage_bytes"`
	StorageQuotaByte int64 `json:"storage_quota_bytes"`
}

type EnterpriseAdministrationQueue struct {
	ContractVersion string                          `json:"contract_version"`
	AsOf            string                          `json:"as_of"`
	Summary         EnterpriseAdministrationSummary `json:"summary"`
	Items           []EnterpriseAdministrationItem  `json:"items"`
}

File: frontend/src/api/enterpriseAdministration.ts
import { get } from '@/utils/request'

export type EnterpriseAdministrationTarget = 'members' | 'knowledge' | 'audit' | 'service_health'
export type EnterpriseAdministrationPriority = 'critical' | 'high' | 'medium'
export type EnterpriseAdministrationItemCode =
  | 'license_or_capability_attention'
  | 'edge_node_offline'
  | 'operating_analysis_access_gap'
  | 'pending_invitations'
  | 'knowledge_processing_failed'
  | 'recent_high_risk_operations'

export interface EnterpriseAdministrationQueueV1 {
  contract_version: 'EnterpriseAdministrationQueueV1'
  as_of: string
  summary: {
    service_level: string
    status: string
    member_quota?: number
    health: string
    member_usage: number
    storage_usage_bytes: number
    storage_quota_bytes: number
  }
  items: Array<{
    code: EnterpriseAdministrationItemCode
    priority: EnterpriseAdministrationPriority
    count: number
    target: EnterpriseAdministrationTarget
  }>
}

export async function getEnterpriseAdministrationQueue(): Promise<EnterpriseAdministrationQueueV1> {
  return (await get('/api/v1/enterprise-administration/queue')) as unknown as EnterpriseAdministrationQueueV1
}
